src/model/ModelMerger.cpp

```diff
@@ -1012,12 +1016,18 @@ namespace model {
     for (const auto& iddObjectType : iddObjectTypesToMerge()) {
       for (auto& currenObject : currentModel.getObjectsByType(iddObjectType)) {
         if (m_currentToNewHandleMapping.find(currenObject.handle()) == m_currentToNewHandleMapping.end()) {
-          currenObject.remove();
+          if ((iddObjectType == IddObjectType::OS_Site) || (iddObjectType == IddObjectType::OS_Facility)
+              || (iddObjectType == IddObjectType::OS_Building)) {
+            // These are unique objects, so no need to delete them
+            continue;
+          } else {
+            currenObject.remove();
+          }
```

Comment:
Actual fix in `ModelMerge::mergeModels`. This works because `getCurrentModelObject` will grab them just fine, and the mergeSite / mergeFacility will deal with it gracefully, only overriding stuff coming from new_Model if it isn't defaulted.

I think this is functionally equivalent to < 3.1.0, and still works when the floorspace comes up with a northAxis for eg